erpnext/stock/doctype/warehouse/warehouse.py

```diff
@@ -199,4 +229,48 @@ def get_warehouses_based_on_account(account, company=None):
 		frappe.throw(_("Warehouse not found against the account {0}")
 			.format(account))
 
-	return warehouses
\ No newline at end of file
+	return warehouses
+
+def get_warehouse_list(doctype, txt, filters, limit_start, limit_page_length=20, order_by="modified"):
+	"""Getting list of warehouse and there items details from Warehouse and Bin doctype."""
+	user = frappe.session.user
+	customer = frappe.get_value("User", user, "full_name")
+	if customer == "Administrator":
+		warehouses = frappe.db.sql('''select name from `tabWarehouse`
+		where published=1''')
+		docs = []
+		for warehouse in warehouses:
+			docs += frappe.db.sql('''select * from `tabBin`
+			where warehouse=%s''', (warehouse), as_dict = True)
+		for doc in docs:
+			item_name = frappe.get_value("Item", doc.item_code, "item_name")
+			doc = doc.update({"item_name": item_name})
+		return docs
+
+	elif customer :
+		docs = frappe.db.sql('''SELECT * FROM `tabBin`
+		WHERE warehouse=(SELECT name FROM `tabWarehouse`
+		WHERE customer=%s AND published=1)''', (customer), as_dict = True)
+		for doc in docs:
+			item_name = frappe.get_value("Item", doc.item_code, "item_name")
+			doc = doc.update({"item_name": item_name})
+		return docs
```

Comment:
- Commonify this.
- There shouldn't be an explicit check for Admin, if there is a customer with admin user, only then it should be visible